samples/python/refresh-authors.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/python
+
+###############################################################################
+# ex. python refresh-authors.py --conf MyConf.org/2017 --baseurl http://localhost:3000
+#       --username admin --password admin_pw
+#
+# To be run to refresh author groups in case of revisions and/or withdrawals
+# Expects config to define BLIND_SUBMISSION, SUBMISSION and AUTHORS as well
+# as an Authors group for each SUBMISSION paper
+###############################################################################
+
+## Import statements
+import argparse
+import sys
+from openreview import *
+
+## Argument handling
+parser = argparse.ArgumentParser()
+parser.add_argument('-c','--conf', required=True, help = "the full path of the conference group ex. MyConf.org/2017")
+parser.add_argument('--baseurl', help="base url")
+parser.add_argument('--username')
+parser.add_argument('--password')
+
+args = parser.parse_args()
+
+## Initialize the client library with username and password
+client = Client(baseurl=args.baseurl, username=args.username, password=args.password)
+
+## check conference directory exists
+base_path = "../../venues/"+args.conf
+config_path = base_path+"/python/"
+if os.path.isfile(config_path+"config.py") is False:
+    print "Cannot locate config.py in:"+config_path
+    sys.exit()
+## load conference specific data
+sys.path.insert(0, config_path)
+import config
+
+### update authors lists
+all_authors = []
+submissions = client.get_notes(invitation=config.SUBMISSION)
+blinded = client.get_notes(invitation=config.BLIND_SUBMISSION)
+for blind_note in blinded:
+    #find associated note in Submissions to get true authors
+    parent_forum = blind_note.original
+    note_authors = []
+    found_profile = False
+    #for parent_notes = submissions[].forum == note.original
+    for note in submissions:
+        if note.forum == parent_forum:
+            # found parent note
+            num_str = str(note.number)
+            paper_id = config.CONF + "/-/Paper"+num_str
+            # convert email ids to twiddle ids where possible
+            for email in note.content['authorids']:
+                email = email.lower()
+                try:
+                    profile = client.get_profile(email)
+                    note_authors.append(profile.id)
+                    found_profile = True
+                    #print "found profile "+email
+                except openreview.OpenReviewException as e:
+                    # no ~id, use the email
+                    note_authors.append(email)
+                    #print "no profile "+email
```

Comment:
I might actually keep some kind of print statement here when an exception gets thrown... something as simple as `print "no profile found: ", e`. The reason I think it's worthwhile is because it's possible for the exception to be something other than the a "not found" error.

either that, or you could check the contents of the error to make sure that it's actually the error you think. If it's not, you can call `raise(e)` and it will throw the error as normal.

Reply:
For any error that you could get from the get_profile call, it should be the email the default value, so I think it is ok.